src/interpreter/binarySearch.ts

```diff
@@ -4,16 +4,16 @@
  */
 
 import {AbsoluteCellRange} from '../AbsoluteCellRange'
-import {InternalScalarValue, simpleCellAddress} from '../Cell'
+import {CellError, simpleCellAddress} from '../Cell'
 import {DependencyGraph} from '../DependencyGraph'
-import {InterpreterValue} from './InterpreterValue'
+import {EmptyValue, getRawValue, RawInterpreterValue, RawScalarValue} from './InterpreterValue'
 
 /*
 * If key exists returns first index of key element in range of sorted values
 * Otherwise returns first index of greatest element smaller than key
 * assuming sorted values in range
 * */
-export function rangeLowerBound(range: AbsoluteCellRange, key: InternalScalarValue, dependencyGraph: DependencyGraph, coordinate: 'row' | 'col'): number {
+export function rangeLowerBound(range: AbsoluteCellRange, key: RawScalarValue, dependencyGraph: DependencyGraph, coordinate: 'row' | 'col'): number {
```

Comment:
It seems `key` type can be narrowed to `RawNoErrorScalarValue`. Then checks for `CellError` are unnecessary  in `compare`